pkg/alertmanager/multitenant.go

```diff
@@ -845,13 +888,82 @@ func (am *MultitenantAlertmanager) GetStatusHandler() StatusHandler {
 	}
 }
 
+// replicateStateForUser attempts to replicate a partial state sent by an alertmanager to its other replicas through the ring.
+func (am *MultitenantAlertmanager) replicateStateForUser(ctx context.Context, userID string, part *clusterpb.Part) error {
```

Comment:
The chain of calls leading to this function is a bit complicated. What if we move this function to `state`? You need to pass `selfAddress`, the ring client (not lifecycler) and the clients pool.

Reply:
If I understand it correctly, moving this to `state` can make it more difficult to implement fairness between different users.